### PR TITLE
[PipelineAdapters] Deprecate unused rundb function argument

### PR DESCRIPTION
**Pull request by liranbg**
Argument was never used, deprecate it

in addition, fix `as_step` outputs argument as it should be a list (and treated as list in its usages)

### CODE REVIEW

File: mlrun/runtimes/base.py [hunk 1]
```diff
@@ -674,7 +674,7 @@ def as_step(
         selector="",
         hyper_param_options: HyperParamOptions = None,
         inputs: dict = None,
-        outputs: dict = None,
+        outputs: list = None,
```
Comment (TomerShor) [inline mlrun/runtimes/base.py:677]: This is a user-facing function, how will this change affect its usage?

**Review by TomerShor — APPROVED** [review]
LGTM, just one question

File: mlrun/runtimes/base.py (hunk 1 above)
Comment (liranbg) [reply mlrun/runtimes/base.py:677]: it won't really change anything as this is type-hinting. if a client passed a dictionary - the execution fails

File: pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/ops.py [hunk 2]
```diff
@@ -441,9 +445,7 @@ def get_default_reg():
     return ""
 
 
-def format_summary_from_kfp_run(
-    kfp_run, project=None, run_db: "mlrun.db.RunDBInterface" = None
-):
+def format_summary_from_kfp_run(kfp_run, project=None):
```
Comment (alonmr) [inline pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/ops.py:448]: 😭  this was needed

**Pull request merged**